test cases/d/12 link directly to lib/meson.build

```diff
@@ -0,0 +1,14 @@
+project('MesonLinkBoostBug', ['d', 'cpp'])
+
+src = ['main.d', 'external.cpp']
+
+boost_dep = dependency('boost', required : false)
```

Comment:
The boost dep will not link to any libraries unless a specific module (which is not header only) is specified. So, this won't link to anything, which is OK in this case because `boost::optional` is header only, but this won't help to reproduce the bug.

Reply:
gah. @#$# header only "libraries"

Reply:
alright, now using threads for absolutely no reason :)